### PR TITLE
IGVF-1545-CLS-ORF-audit

### PR DESCRIPTION
**Pull request by khzlin**

### CODE REVIEW

File: src/igvfd/audit/construct_library_set.py [hunk 1]
```diff
@@ -118,3 +118,34 @@ def audit_construct_library_set_guide_design(value, system):
...
             yield AuditFailure('missing guide RNA sequences', f'{detail} {description}', level='NOT_COMPLIANT')
+
+
+@audit_checker('ConstructLibrarySet', frame='object')
+def audit_construct_library_set_orf_gene(value, system):
+    '''
+    [
+        {
+            "audit_description": "Genes listed in the library are expected to match the open read frame gene.",
+            "audit_category": "inconsistent gene",
```
Comment (ian-whaling) [inline src/igvfd/audit/construct_library_set.py:129]: I think `inconsistent genes` is more consistent with category naming guidelines.

File: src/igvfd/audit/construct_library_set.py [hunk 2]
```diff
@@ -118,3 +118,34 @@ def audit_construct_library_set_guide_design(value, system):
...
+        for o in orf_ids:
+            orf_object = system.get('request').embed(o + '@@object?skip_calculated=true')
+            if 'gene' in orf_object:
+                for d in orf_object['gene']:
+                    orf_genes.add(d)
+
+        if orf_genes != library_genes:
+            detail = (
+                f'ConstructLibrarySet {audit_link(path_to_text(value["@id"]),value["@id"])} '
+                f'has a `small_scale_gene_list` which does not match the `gene` of its associated `orf_list`.'
```
Comment (ian-whaling) [inline src/igvfd/audit/construct_library_set.py:149]: Don't use camelcase for item names, but instead sentence case. i.e. ConstructLibrarySet -> Construct library set. Also I would reserve the code formatting for field names not item names so theres no need for backticks on gene.

**Review by ian-whaling — CHANGES_REQUESTED** [review]
Some rewording requests.

File: src/igvfd/audit/construct_library_set.py (hunk 1 above)
Comment (khzlin) [reply src/igvfd/audit/construct_library_set.py:129]: done

File: src/igvfd/audit/construct_library_set.py (hunk 2 above)
Comment (khzlin) [reply src/igvfd/audit/construct_library_set.py:149]: done

File: src/igvfd/audit/construct_library_set.py [hunk 3]
```diff
@@ -145,7 +145,7 @@ def audit_construct_library_set_orf_gene(value, system):
 
         if orf_genes != library_genes:
             detail = (
-                f'ConstructLibrarySet {audit_link(path_to_text(value["@id"]),value["@id"])} '
-                f'has a `small_scale_gene_list` which does not match the `gene` of its associated `orf_list`.'
+                f'Construct library set {audit_link(path_to_text(value["@id"]),value["@id"])} '
+                f'has a `small_scale_gene_list` which does not match the genes of its associated `orf_list`.'
             )
             yield AuditFailure('inconsistent gene', f'{detail} {description}', level='ERROR')
```
Comment (ian-whaling) [inline src/igvfd/audit/construct_library_set.py:151]: The audit failure should reflect the change in the docstring.

Comment (khzlin) [reply src/igvfd/audit/construct_library_set.py:151]: thanks for catching that

**Review by ian-whaling — APPROVED** [review]

**Pull request merged**